### DISCUSSION TITLE
MacOS X 12 Monterey and piwebcam

### DISCUSSION THREAD

**Discussion by RTHPJM** [geerlingguy/pi-webcam#52] [General]
Hi Jeff,

You’re a Mac guy. Have you tried the showmewebcam (piwebcam) on Monterey yet? I just installed the released version of Monterey on my Ryzentosh, and I’ve lost the connection to the camera. My very brief testing results in a conclusion that Monterey has stopped seeing the piwebcam as a UVC device.
I can boot into Big Sur and Monterey with the same OpenCore EFI (there are some Wi-Fi/Bluetooth kexts that differ on min/max kernel settings, but that’s all). Big Sur sees the camera as UVC, but Monterey just shows the device as USB, no attached UVC.
There are some sparse Google hits of UVC external cameras going AWOL during Beta phases, but nothing overwhelming.

I just wondered if you have tested the combination and what your results might be?…

Cheers
Paul

**Comment by geerlingguy** [comment]
I haven't yet upgraded to Monterey just because I don't do it for a few weeks as a general rule :)

**Reply by RTHPJM** [reply]
I popped over to the showmewebcam GitHub repo. someone had already logged an issue. I've added a comment, in the process realised that MacOS Monterey is still seeing the webcam as a UVC device from a IO Registry perspective, but the OS is not picking it up a a video device. Looks like Apple are not being a flexible with generic external UVC devices. Maybe we need a DEXT or codeless KEXT to match the VID and PID (I'm out of my comfort zone at this point, but might see it as an opportunity to learn something new!).

Or maybe it's just a bug with the new Dext and Camera Assistant plugins???

**Comment by stelterlab** [comment]
I have the same problem on my MacBook Pro which got Monterey for testing. The pi webcam is listed in the system report, but not recognized as Webcam in the usual apps.
But I use a USB-C hub as connection for the pi webcam to the MBP. I will try it with a simple USB-C Adapter as soon as possible. But I'm afraid its a Monterey feature.

**Reply by stelterlab** [reply]
No difference with the direct connected pi webcam. So the hub was not the culprit. There are reports from differents Apple/News sites incoming that quite some usb devices are not working properly under Monterey.
